File: backend/src/main/java/com/accesodatos/service/MatchService.java
package com.accesodatos.service;

import java.time.LocalDate;
import java.util.List;

import com.accesodatos.dto.match.MatchRequestDto;
import com.accesodatos.dto.match.MatchResponseDto;
import com.accesodatos.dto.match.MatchResultRequestDto;

public interface MatchService {

	List<MatchResponseDto> getAllMatches();

	MatchResponseDto getMatchById(Long matchId);

	MatchResponseDto createMatch(MatchRequestDto matchRequestDto);

	MatchResponseDto updateMatch(Long matchId, MatchRequestDto matchRequestDto);

	void deleteMatch(Long matchId);

	List<MatchResponseDto> getMatchesByDateOrderByCompetition(LocalDate date);

	void updateIsPlayingMatch();

	List<MatchResponseDto> getMatchesReadyToValidate();

	MatchResponseDto validateMatch(Long matchId, MatchResultRequestDto MatchResult);
}	

File: frontend/src/service/match.service.ts
import { MatchRequest } from "../types/Match";
import api from "./api";

const getMatchesByDate = async (date: string) => {
  const response = await api.get(`/matches/by-date`, {
    params: { date },
  });
  return response.data.data;
};

const getMatchById = async (matchId: number) => {
  const response = await api.get(`/matches/${matchId}`);
  return response.data;
};

const createMatch = async (match: MatchRequest) => {
  const response = await api.post(`/matches`, match);
  return response.data;
};

const updateMatch = async (matchId: number, match: MatchRequest) => {
  const response = await api.put(`/matches/${matchId}`, match);
  return response.data;
};

const getMatchesReadyToValidate = async () => {
  const response = await api.get(`/matches/ready`);
  return response.data.data;
};

const validateMatch = async (matchId: number, result: string) => {
  const response = await api.put(`/matches/${matchId}/validate`, {
    result,
  });
  return response.data;
};

const MatchService = {
  getMatchesByDate,
  getMatchById,
  createMatch,
  updateMatch,
  getMatchesReadyToValidate,
  validateMatch,
};

export default MatchService;
